python/lsst/ts/m2gui/controltab/tab_diagnostics.py

```diff
@@ -33,7 +44,689 @@ class TabDiagnostics(TabDefault):
+            "J1-W12-1 Communication Power Breaker OK",
+            "Spare Input 16",
+            "Spare Input 17",
+            "Spare Input 18",
+            "Spare Input 19",
+            "Spare Input 20",
+            "Spare Input 21",
+            "Spare Input 22",
+            "Spare Input 23",
+            "J1-W12-2 Communication Power Breaker OK",
+            "J2-W13-1 Communication Power Breaker OK",
+            "J2-W13-2 Communication Power Breaker OK",
+            "J3-W14-1 Communication Power Breaker OK",
+            "J3-W14-2 Communication Power Breaker OK",
+            "Spare Input 29",
+            "Spare Input 30",
+            "Interlock Power Replay On",
+        ]
+
+    def _get_list_digital_status_output(self):
+        """Get the list of digital output status from the bit 0 to 7.
+
+        Returns
+        -------
+        `list`
+            List of the digital output status.
+        """
+        return [
+            "Motor Power",
+            "Communication Power",
+            "Interlock Enable",
+            "Reset Motor Breakers",
+            "Reset Communication Breakers",
+            "Spare Output 5",
+            "Spare Output 6",
+            "Spare Output 7",
+        ]
+
+    def _create_controls_digital_status(self, number):
+        """Create the controls of digital status.
+
+        Parameters
+        ----------
+        number : `int`
+            Number of controls.
+
+        Returns
+        -------
+        controls : `list [PySide2.QtWidgets.QPushButton]`
+            List of the controls of digital status.
+        """
+
+        controls = list()
+        for idx in range(number):
+            control = set_button(
+                "", self._callback_control_digital_status, idx, is_checkable=True
+            )
+            self._update_control_text(control, False)
+
+            controls.append(control)
+
+        return controls
+
+    @Slot()
+    def _callback_control_digital_status(self, idx):
+        """Callback of the digital-status-control button.
+
+        This allows the user to command individual binary signals to the cell
+        regarding breaker resets, individual power supply controls, or
+        interlocks.
+
+        Parameters
+        ----------
+        idx : `int`
+            Bit index.
+        """
+
+        control = self._digital_status_control[idx]
+        is_checked = control.isChecked()
+
+        is_successful = run_command(self.model.set_bit_digital_status, idx, is_checked)
+
+        if is_successful:
+            self._update_control_text(control, is_checked)
+        else:
+            control.setChecked(not is_checked)
+
+    def _update_control_text(self, control, is_checked):
+        """Update the text of control.
+
+        Parameters
+        ----------
+        control : `PySide2.QtWidgets.QPushButton`
+            Control.
+        is_checked : `bool`
+            Control is checked or not.
+        """
+
+        if is_checked:
```

Comment:
This is a great idea if I choose the inheritance instead of composition to reuse the functions. However, I prefer to use the inheritance only when it is required. I felt the Ot uses the inheritance pretty much already and I tried to avoid to be involved in that. I will keep your suggestion in mind in the future's refactoring. Thanks!